PetAdoption-iOS/PetAdoption-iOS/PetListingDetailViewController.swift

```diff
@@ -7,150 +7,144 @@
+    ////////////////////////////////////////////////////////////
+    // MARK: - Properties
+    ////////////////////////////////////////////////////////////
 
-    /*
-    // MARK: - Navigation
+    let cellIdentifier = "Cell"
+    var pet : PTKPet!
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
-    }
-    */
+    ////////////////////////////////////////////////////////////
+    // MARK: - View Controller Life Cycle
+    ////////////////////////////////////////////////////////////
 
-}
+    override func viewDidLoad()
+    {
+        // TODO: The Call, Email, and Website controls under the images will still need to be implemented.
 
-extension PetListingDetailViewController : UITableViewDelegate {
-	
-}
+        super.viewDidLoad()
+        self.title = self.pet.name
+        self.pageControl.numberOfPages = self.pet.imageURLPaths.count
+        self.titleLabel.text = self.pet.name
 
-extension PetListingDetailViewController : UITableViewDataSource {
-	func numberOfSectionsInTableView(tableView: UITableView) -> Int {
-		return 1;
-	}
-	
-	func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-		return self.pet.petAttributes.count;
-	}
-	
-	func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-		let cell = tableView.dequeueReusableCellWithIdentifier(cellIdentifier);
-		let data = self.pet.petAttributes[indexPath.row];
-		
-		cell!.textLabel?.text = data.attributeTitle;
-		cell!.textLabel?.backgroundColor = UIColor.clearColor();
-		cell!.detailTextLabel?.text = data.attributeValue;
-		cell!.detailTextLabel?.backgroundColor = UIColor.clearColor();
+        let shelter = self.pet.petShelter
+
+        self.generalInfoView.addTopBorder(color: UIColor.darkGrayColor())
+        self.genderLabel.text = self.pet.gender.rawValue
+        self.sizeLabel.text = self.pet.size
+        self.ageLabel.text = self.pet.age.description
+        self.locationLabel.text = shelter.name == "" ? "N/A" : shelter.name
+
+        self.descriptionView.addTopBorder(color: UIColor.darkGrayColor())
+        self.descriptionLabel.text = self.pet.description
+
+        self.contactInfoView.addTopBorder(color: UIColor.darkGrayColor())
+        self.shelterNameLabel.text = shelter.name
+        self.shelterAddress1Label.text = shelter.address1
+        if let address2 = shelter.address2
+        {
+            self.shelterAddress2Label.hidden = false
+            self.shelterAddress2Label.text = address2
+        }
+        else
+        {
+            self.shelterAddress2Label.hidden = true
+        }
+
+        let isCityStateZipBlank = (shelter.city == "") || (shelter.state == "") || (shelter.zipcode == "")
```

Comment:
I'll change the fields in the shelter class to allow nil so this is easier. Also we should add a function that creates a localised readable string of the shelter's details so the viewController doesn't have to know how to do it.